doc/host_config_schema/cchost_config.json

```diff
@@ -250,10 +265,11 @@
           "minimum": 1
         }
       },
-      "description": "This section includes configuration for the node x509 identity certificate"
+      "description": "This section includes configuration for the node x509 identity certificate",
+      "additionalProperties": false
     },
     "node_data_json_file": {
-      "type": "string",
+      "type": ["string", "null"],
```

Comment:
I don't understand the purpose of these changes. Do we want/expect `"type": null` entries in the JSON config? I would have thought these were optional fields, but if present they have to be string?

Reply:
We set `node_data_json_file` to `null` in the Python infra because of the [Jinja template we currently use](https://github.com/microsoft/CCF/blob/main/tests/config.jinja#L17). We could modify the template to not include the field when it's `null` but I thought allowing a `null` value for all optional configuration entries that refer to files give a little more flexibility.

Reply:
In general, I think treating both missing keys and `"key": null` as valid is an unnecessary complication, and not flexibility that we want. It makes manually type-checking and parsing the doc more complicated. That said, maybe it's the right choice in this instance, as it makes the generated configs that deliberately omit this value more explicit.